packages/fuel-indexer-schema/src/db/tables.rs

```diff
@@ -419,6 +424,81 @@ impl Schema {
     }
 }
 
+fn get_foreign_keys(schema: &str) -> HashMap<String, HashMap<String, String>> {
+    let mut foreign_keys: HashMap<String, HashMap<String, String>> = HashMap::new();
+
+    let base_ast = match parse_schema::<String>(BASE_SCHEMA) {
+        Ok(ast) => ast,
+        Err(e) => {
+            panic!("Error parsing graphql schema {e:?}",)
+        }
+    };
+    let (primitives, _) = build_schema_objects_set(&base_ast);
+
+    let ast = match parse_schema::<String>(schema) {
+        Ok(ast) => ast,
+        Err(e) => panic!("Error parsing graphql schema {e:?}",),
+    };
+    let types_map = build_schema_fields_and_types_map(&ast);
+
+    for def in ast.definitions.iter() {
```

Comment:
feel like this logic related to definitions could also be put into it's own function. So that the broad functionality of `get_foreign_keys()` can be digested more or less at a glance. `process_defenitions()`?

Reply:
Yeah, I'm down for that, it is a bit unwieldy.

Reply:
Agree with @rvmelkonian here

Reply:
Also `get_foreign_keys` and `process_definitions` seem like perfect candidates for simple unit tests.